src/literal/imp.rs

```diff
@@ -645,7 +645,7 @@ impl BoyerMooreSearch {
     }
 
     fn len(&self) -> usize {
-        return self.pattern.len();
```

Comment:
I think this is actually the only change in this PR that I agree with. Could you remove the rest please?